QEfficient/lora/auto.py

```diff
@@ -0,0 +1,396 @@
+        ), "Not all adapters have the same ranks"
+
+        # set active adapter id to current max
+        self.active_adapter_to_id[adapter_name] = self.max_num_adapters
+
+        # add active adapter to set
+        self.active_adapters.add(adapter_name)
+        self.max_num_adapters = len(self.active_adapters)
+
+        return self.active_adapter_to_id[adapter_name]
+
+    def delete_adapter(self, adapter_name: str):
+        if adapter_name not in self.active_adapters:
+            print(f"Adapter name {adapter_name} is not set active yet")
+            return False
+
+        self.active_adapters.discard(adapter_name)
+        self.max_num_adapters -= 1
+        self.active_adapter_to_id.pop(adapter_name)
+
+        # renumbering of active adapter id
+        for index, (key, value) in enumerate(self.active_adapter_to_id.items()):
+            self.active_adapter_to_id[key] = index
+
+        logger.warning(f"Deleting {adapter_name} from active adapters.")
+        if self.onnx_path or self.qpc_path:
+            logger.warning("Please redo compile_and_export() to reflect the active adapters changes.")
+
+        return True
+
+    def get_adapter_id(self, adapter_name):
+        "get the adapter_id that maps to the adapter_name"
+
+        return self.active_adapter_to_id[adapter_name]
+
+    def load_adapter_weights_to_model(self):
+        "Loads adapter weights to the model's multilora layer in a stacked format"
+
+        num_hidden_layers = len(self.model.model.layers)
+        for i in range(num_hidden_layers):
+            for target_module in self.target_modules_for_all_adapters:
+                # stack all adapters weights
+                a_tensor_list = list(range(self.max_num_adapters))
+                b_tensor_list = list(range(self.max_num_adapters))
+                c_tensor_list = list(range(self.max_num_adapters))
+
+                for lora_name, lora_id in self.active_adapter_to_id.items():
+                    if (
+                        target_module == "q_proj"
+                        or target_module == "k_proj"
+                        or target_module == "v_proj"
+                        or target_module == "up_proj"
+                    ):
+                        a_tensor_list[lora_id] = torch.from_numpy(
+                            self.adapter_weights[lora_name][
+                                f"base_model.model.model.layers.{i}.self_attn.{target_module}.lora_A.weight"
+                            ]
+                        )
+                        b_tensor_list[lora_id] = torch.from_numpy(
+                            self.adapter_weights[lora_name][
+                                f"base_model.model.model.layers.{i}.self_attn.{target_module}.lora_B.weight"
+                            ]
+                        )
+                    elif target_module == "up_proj" or target_module == "gate_proj" or target_module == "down_proj":
```

Comment:
remove this elif for up_proj/gate_proj/down_proj

Reply:
removed